json-format.md

```diff
@@ -129,34 +129,82 @@ OPTIONAL not omitted attributes MAY be represeted as a `null` JSON value.
 
 ### 3.1. Handling of "data"
 
+The JSON representation of the event "data" payload is determined by the runtime
+type of the `data` content and the value of the [`datacontenttype`
+attribute][datacontenttype].
+
+#### 3.1.1. Payload Serialization
+
 Before taking action, a JSON serializer MUST first determine the runtime data
 type of the `data` content.
 
 If the implementation determines that the type of data is `Binary`, the value
 MUST be represented as a [JSON string][json-string] expression containing the
 [Base64][base64] encoded binary value, and use the member name `data_base64` to
-store it inside the JSON object.
-
-For any other type, the implementation MUST translate the data value into a
-[JSON value][json-value], and use the member name `data` to store it inside the
-JSON object.
-
-Out of this follows that use of the `data` and `data_base64` members is mutually
-exclusive in a JSON serialized CloudEvent.
+store it inside the JSON representation. The `datacontenttype` SHOULD reflect
+the format of the original binary data.
+
+If the type of data is NOT `Binary`, the implementation MUST next determine
+whether the value of the `datacontenttype` attribute is _JSON-oriented_. A
+JSON-oriented content type is one whose subtype is `json` or has a `+json`
+format extension—that is, if it has the form `*/json` or `*/*+json`.
```

Comment:
Does anyone know of more standard terminology and/or reference links we could use for this?

Reply:
I think we should limit it to json, unless jsonc is also allowed for JSON format overall (e.g. `application/cloudevents+jsonc`) which it isn't currently.

As for `application/json+somethingelse`, I don't know. That seems not to make sense to me conceptually (a JSON document in a different, supposedly non-json, format?) But if that actually exists, maybe we should include it for completeness.

Reply:
That seems reasonable. Though I've been trying to avoid using the word "format" to avoid confusion with the fact that overall this is a "json-format". That is, to draw a clear distinction between the fact that the overall envelope is JSON, and whether or not the content type also implies JSON data.

Reply:
Thanks for the context @dazuma.  I can see that the distinction that we're talking about the data attribute's value specifically is not as explicit as it could be.  I think I'd try to add that greater specificity rather than establish new terminology.  Maybe we can get away with "declares the data attribute to contain JSON formatted content"?  That surely a mouthful.  You have my sympathy, these language crafting niggles can be the hardest.

Reply:
Thanks @erikerikson. I went ahead and used your more descriptive language, and I think it does work better.